subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/DefaultVersionedComponentChooser.java

```diff
@@ -68,51 +68,55 @@ private boolean isGeneratedModuleDescriptor(ComponentResolveMetadata componentRe
         return componentResolveMetadata.isGenerated();
     }
 
-    public void selectNewestMatchingComponent(Collection<? extends ModuleComponentResolveState> versions, BuildableComponentSelectionResult result, ModuleVersionSelector requested) {
+    public void selectNewestMatchingComponent(Collection<? extends ModuleComponentResolveState> versions, ComponentSelectionContext result, ModuleVersionSelector requested) {
```

Comment:
Instead of adding comments, I think it would make this more readable if it was broken down into smaller pieces.